### ISSUE TITLE
error MSB6006: "rtcli.exe" exited with code -1073741571 + [TsProperty] not working

### ISSUE THREAD

**Issue by vmandic** [reinforced/Reinforced.Typings#53]
Hello! 

I have a .csproj with circa 250 C# files (a mix of classes, interfaces and enums). I was able to successfully run your translator tool for interfaces and enums, but when including all the classes then I get the error from the title:

> packages\Reinforced.Typings.1.3.7\build\Reinforced.Typings.targets(76,3): error MSB6006: "rtcli.exe" exited with code -1073741571.

Could you please help me out on how to narrow down the issue, so I could potentially locate the problematic class and see what's bugging the translator tool or even better is this a sort of a common issue for which you can help me out?

Currently I am narrowing it down by filtering on namespaces as when I cherry pick some classes the translation is successful.

I'm using a custom configure method per instructions on wiki, like this:

```
public static void Configure(ConfigurationBuilder builder)
		{

// "TranslationConfiguration" below, is the name of this static class which has this method, and 
// which is in the assembly that holds all the view models

			var asm = Assembly.GetAssembly(typeof(TranslationConfiguration));
			var viewModelTypes = asm.GetTypes();

			var classes = new List<Type>();
			var interfaces = new List<Type>();
			var enums = new List<Type>();

			foreach (var t in viewModelTypes)
			{
				if (t.IsClass) classes.Add(t);
				else if (t.IsInterface) interfaces.Add(t);
				else if (t.IsEnum) enums.Add(t);
			}

			var classesConfig = new Action<IClassConfigurationBuilder>(c => c.WithAllProperties());
			var intfConfig = new Action<IInterfaceConfigurationBuilder>(c =>
			{
				c.WithAllFields();
				c.WithAllProperties();
				c.AutoI(false);
			});

			builder.ExportAsClasses(classes, classesConfig);
			builder.ExportAsInterfaces(interfaces, intfConfig);
		        builder.ExportAsEnums(enums);
		}
```

Hope you can advise me how to debug the rtcli or any ideas would be welcome, thanks.

Thanks for making this great tool.
V.

**Comment by pavel-b-novikov** [comment]
See here: https://github.com/reinforced/Reinforced.Typings/issues/35

**Comment by vmandic** [comment]
Thanks, this helped me sort out a bit the error which is the infamous StackOverflowException, here is a part of the trace of the stack where it seems to happen after processing my class (lets call it Class3) Class3

> ...
> Exported REDACTED.Search.Class1
> Exported REDACTED.Search.Class2
> Exported REDACTED.Search.Class3
> 
> Process is terminated due to StackOverflowException.
> C:\Users\vmandic\Documents\Projects\REDACTED\packages\Reinforced.Typings.1.3.7\build\Reinforced.Typings.targets(76,3): error MSB6006: "rtcli.exe" exited with code -1073741571. [C:\Users\vmandic\Documents\Projects\REDACTED\ViewModels\ViewModels.csproj]
> Done executing task "RtCli" -- FAILED.
> Done building target "ReinforcedTypingsGenerate" in project "ViewModels.csproj" -- FAILED.
> Done executing task "CallTarget" -- FAILED.
> Done building target "CoreCompile" in project "ViewModels.csproj" -- FAILED.
> Target "_CheckForCompileOutputs" in file "C:\Program Files (x86)\MSBuild\14.0\bin\amd64\Microsoft.Common.CurrentVersion.targets" from project "C:\Users\vmandic\Documents\Projects\REDACTED\ViewModels\ViewModels.csproj" (target "_CleanGetCurrentAndPriorFileWrites" depends on it):
> Done building target "_CheckForCompileOutputs" in project "ViewModels.csproj".
> Target "_SGenCheckForOutputs" skipped, due to false condition; ('$(_SGenGenerateSerializationAssembliesConfig)' == 'On' or ('@(WebReferenceUrl)'!='' and '$(_SGenGenerateSerializationAssembliesConfig)' == 'Auto')) was evaluated as ('Off' == 'On' or (''!='' and 'Off' == 'Auto')).
> Target "_CleanGetCurrentAndPriorFileWrites" in file "C:\Program Files (x86)\MSBuild\14.0\bin\amd64\Microsoft.Common.CurrentVersion.targets" from project ...

So... I was able to narrow it down to a property of a Class4 (coming after Class3...) which is:
`public Newtonsoft.Json.Linq.JToken Value { get; set; }` ... [TsIgnore] solved the issue, but trying to override the type with [TsProperty(Type="any")] results in the same StackOverflowException.

Any help or ideas?

**Comment by pavel-b-novikov** [comment]
Yup, some types are Enumerable, but actually arent. I have encountered the same error with ```JRaw``` from Json.NET. So be careful.

**Comment by vmandic** [comment]
Thanks for the feedback, fixed it with the fluent `builder.Substitute() `(real handy!) config method.
The `[TsProperty(Type="any")]` did not work in my case at all, I tried it on some other props and it is just like it is ignored, while the Configure method configs pass and are OK. Do you have TsProperty unit test covered?

**Comment by vmandic** [comment]
Also tried: `[TsProperty(StrongType = typeof(string))] public int SomeIntToStrProp { get; set; }` in a new project, seems it is skipped, TsIgnore working fine.

**Comment by pavel-b-novikov** [comment]
That is bug, but there must be test for it. I will check soon

**Comment by pavel-b-novikov** [comment]
Problem was identified and fixed in 1.4.1
The case was conflict between Fluent and attribute configuration. E.g. when you use ```.WithPublicProperties``` along with ```[TsProperty]```
